### PR TITLE
Introduce optional dependencies for package

### PR DESCRIPTION
**Pull request by penguinolog**
Simplify package installation with extras
Note `trio` and `ZeroMQ` event loop

##### Checklist
- [ ] I've ensured that similar functionality has not already been implemented
- [ ] I've ensured that similar functionality has not earlier been proposed and declined
- [X] I've branched off the `master` or `python-dual-support` branch
- [X] I've merged fresh upstream into my branch recently
- [X] I've ran `tox` successfully in local environment
- [X] I've included docstrings and/or documentation and/or examples for my code (if this is a new feature)

Fix: #441

### CODE REVIEW

**Comment by penguinolog** [comment]
Note: this PR not introduces normal "requirements"

File: README.rst [hunk 1]
```diff
@@ -31,6 +31,13 @@ To install using pip
    
    pip install urwid
 
+For advanced functionality extra requirements need to be installed.
+Example for ZeroMQ event loop and LCD display:
```
Comment (wardi) [inline README.rst:35]: is there a standard way to get a list of optional dependencies and their descriptions or should we add that here/in the docs?

Comment (ulidtko) [reply README.rst:35]: They're described in pyproject.toml... No, don't duplicate in docs, will get outdated

Comment (penguinolog) [reply README.rst:35]: Pip feature request is multiple years old, even solution with API is already added to the issue.
At this moment optional dependencies are not visible without API

**Review by wardi — APPROVED** [review]

**Pull request merged**